src/shogun/machine/KernelMachine.cpp

```diff
@@ -553,72 +503,57 @@ SGVector<float64_t> CKernelMachine::apply_locked_get_output(
 		io->disable_progress();
 
 	/* custom kernel never has batch evaluation property so dont do this here */
-	// TODO: port to use OpenMP backend instead of pthread
-#ifdef HAVE_PTHREAD
-	int32_t num_threads=parallel->get_num_threads();
-#else
-	int32_t num_threads=1;
-#endif
-	ASSERT(num_threads>0)
-
-	if (num_threads<2)
-	{
-		S_THREAD_PARAM_KERNEL_MACHINE params;
-		params.kernel_machine=this;
-		params.result=output.vector;
-
-		/* use the parameter index vector */
-		params.start=0;
-		params.end=num_inds;
-		params.indices=indices.vector;
-		params.indices_len=indices.vlen;
-
-		params.verbose=true;
-		apply_helper((void*) &params);
-	}
-#ifdef HAVE_PTHREAD
-	else
+	int32_t num_threads;
+	int64_t step;
+	#pragma omp parallel shared(num_threads, step)
 	{
-		pthread_t* threads = SG_MALLOC(pthread_t, num_threads-1);
-		S_THREAD_PARAM_KERNEL_MACHINE* params=SG_MALLOC(S_THREAD_PARAM_KERNEL_MACHINE, num_threads);
-		int32_t step= num_inds/num_threads;
-
-		int32_t t;
-		for (t=0; t<num_threads-1; t++)
+#ifdef HAVE_OPENMP
+		#pragma opm single
```

Comment:
mmm, I think so. I compile it under the environment with openmp. I don't know why there's no error comes out. I will fix it right now :)